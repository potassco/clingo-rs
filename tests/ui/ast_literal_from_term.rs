// ast_literal_from_term
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let atom = ast::symbolic_atom(term).unwrap();
    let lit = ast::basic_literal_from_symbolic_atom(&loc, ast::Sign::NoSign, atom).unwrap();

    let term2: ast::Term = term;
    let _end = (lit, term2);
}
