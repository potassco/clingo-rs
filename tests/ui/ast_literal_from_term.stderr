error[E0382]: use of moved value: `term`
  --> tests/ui/ast_literal_from_term.rs:11:28
   |
7  |     let term: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
   |         ---- move occurs because `term` has type `Term<'_>`, which does not implement the `Copy` trait
8  |     let atom = ast::symbolic_atom(term).unwrap();
   |                                   ---- value moved here
...
11 |     let term2: ast::Term = term;
   |                            ^^^^ value used here after move
   |
help: consider cloning the value if the performance cost is acceptable
   |
8  |     let atom = ast::symbolic_atom(term.clone()).unwrap();
   |                                       ++++++++
